src/widget/input.rs

```diff
@@ -51,6 +52,7 @@ pub enum Event {
 pub struct Input<'a, Message> {
     id: Id,
     buffer: Buffer,
+    users: Vec<User>,
```

Comment:
Could this borrow a slice to avoid the `clone` in `view`?

```suggestion
    users: &'a [User],
```

Reply:
Unfortunately not. We're currently depending on the IRC crate for managing the user list per channel and it doesn't expose a way to get a reference to this data.

Ideally we manage channels and users ourselves so we could make this optimization. It'd definitely benefit us in the long run as we have better control over our data.

Reply:
A middle ground solution would be to extract channel and user lists after receiving each batch of IRC messages. This will ensure they're in sync but push those types into our data so we could do this.

Reply:
Yup, just took a look at the `irc` crate. A bit unfortunate it relies heavily on interior mutation :/

Reply:
I've created a tracking issue for this here:

https://github.com/squidowl/halloy/issues/102#issue-1790818159

Reply:
Fixed in 3a54355af9121a9a8fc50fee815f394c4bdb537f after rebasing to #104